cl/citations/annotate_citations.py

```diff
@@ -50,7 +51,7 @@ def generate_annotations(
             ]
         else:  # If successfully matched...
             annotation = [
-                f'<span class="citation" data-id="{opinion.pk}"><a href="{opinion.cluster.get_absolute_url()}">',
+                f'<span class="citation" data-id="{html.escape(str(opinion.pk))}"><a href="{html.escape(opinion.cluster.get_absolute_url())}">',
```

Comment:
`str()` necessary for `option.pk` pushes this into overzealous territory doesn’t it? (I was surprised that `escape` doesn’t coerce the argument to a string itself).

Reply:
opinion.pk will always be an int. Why do this? 

`get_absolute_url` will be a `SlugField` with some numbers. Do we really need to escape it?

Reply:
Yea, as you say – both (especially `pk`) are known to contain characters that would cause problems; the argument for it really cultural rather than technical. Routine usage makes it more likely that implementers will do the right thing when they copy paste and make failures to escape more obtrusive. 

But this uncompromising approach only pays off when exceptions are rare and that is very much not CourtListener’s situation today. If the type cast wasn’t necessary, I’d probably stand by the argument for pro forma escaping (since its harmless), but as it is now the extra complexity just raises questions without meaningfully moving the needle on encoding rigor. Gonna drop this commit.